fix: logger-not-working-in-plugins

AnnaHariprasad5123 opened the pull request:
Fixes #36073 

Hi @NilanshBansal 

**Issue :**

**Missing Logging Implementation :**

- Without a logging implementation (such as SLF4J Simple or Logback) in the project's classpath, the logging statements in the plugins cannot be executed.
- As a result, no log output is being printed to the terminal.

**Solution :**

The solution is to add a logging implementation to the plugins parent pom. In this case, you can add the slf4j-simple dependency to your pom.xml file. This will provide a simple logging implementation that will output log statements to the console.

```
<dependency>
<groupId>org.slf4j</groupId>
<artifactId>slf4j-api</artifactId>
<version>1.7.36</version>
</dependency>

<dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-simple</artifactId>
    <version>1.7.36</version> 
</dependency>
```
**Explanation:**

- slf4j-api provides the SLF4J API, which is the interface for logging.
  provides the SLF4J API, which is the interface for logging.
- slf4j-simple provides a simple implementation of the SLF4J API, which is responsible for actually printing the log messages to the console.

**Screenshots :** 

Amazon S3 Plugin and Postgres Plugin 
![image](https://github.com/user-attachments/assets/d3e90b96-2b02-493d-8ffa-44e2aa348fc1)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Enhanced logging capabilities with the integration of SLF4J API and SLF4J Simple implementations.

- **Improvements**
	- Improved log management and output formatting for better monitoring and debugging across various plugins, transitioning from standard output to structured logging.
	- Refined logging practices in multiple plugins to support better maintainability and performance.
	- Removed method for console logging from the Stopwatch class to streamline logging practices.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

NilanshBansal commented:
@AnnaHariprasad5123 thanks for this fix!
Can you please also update the System.out.println() statements to log.debug() in the Plugins. 
This was recently changed in this PR https://github.com/appsmithorg/appsmith/pull/36077

AnnaHariprasad5123 commented:
Hi @NilanshBansal, I replaced all System.out.println to log.debug. Could you check once.

AnnaHariprasad5123 commented:
Hi @NilanshBansal, Could you check once. Let me know if any changes required.

NilanshBansal commented:
@AnnaHariprasad5123 can you also remove the method stopAndLogTimeInMillisWithSysOut() from Stopwatch.java and also replace all the usages of this method with stopAndLogTimeInMillis(), since it was added as a temporary fix
https://github.com/appsmithorg/appsmith/pull/36077/files#diff-8005b0d844e9004cc03abebe5ec522c3ff8be090da2c9288344a5c2a76e4717b

<img width="1077" alt="image" src="https://github.com/user-attachments/assets/07282080-6544-49f6-9cc5-aa23a8090d22">

AnnaHariprasad5123 commented:
Hi @NilanshBansal, I replaced all of them with with stopAndLogTimeInMillis(). Could you check now.

NilanshBansal commented:
@AnnaHariprasad5123 I tried pulling and executing the changes locally, but I am not seeing some logs from PostgresPlugin.java.
For example this log is not getting printed.
https://github.com/appsmithorg/appsmith/blob/55f428c1b68c80ec1a714ade5657bd570487a4ec/app/server/appsmith-plugins/postgresPlugin/src/main/java/com/external/plugins/PostgresPlugin.java#L247
Can you check the reasons for this.

AnaghHegde commented on app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:
```diff
@@ -923,15 +923,14 @@ public Mono<AmazonS3> datasourceCreate(DatasourceConfiguration datasourceConfigu
                         .onErrorResume(exception -> {
-                            System.out.println("Error closing S3 connection.");
-                            exception.printStackTrace();
+                            log.debug("Error closing S3 connection.", exception);
```
```suggestion
                            log.error("Error while closing S3 connection. ", exception.getMessage());
```

AnaghHegde commented on app/server/appsmith-plugins/databricksPlugin/src/main/java/com/external/plugins/DatabricksPlugin.java:
```diff
@@ -108,8 +108,7 @@ public Mono<ActionExecutionResult> execute(
                             // This should not happen ideally.
-                            System.out.println(
-                                    "Error checking validity of Databricks connection : " + error.getMessage());
+                            log.debug("Error checking validity of Databricks connection : " + error.getMessage());
```
```suggestion
                            log.error("Error checking validity of Databricks connection : " + error.getMessage());
```

AnaghHegde commented on app/server/appsmith-plugins/databricksPlugin/src/main/java/com/external/plugins/DatabricksPlugin.java:
```diff
@@ -303,14 +301,14 @@ public Mono<Connection> datasourceCreate(DatasourceConfiguration datasourceConfi
             } catch (SQLException e) {
                 // This should not happen ideally.
-                System.out.println("Error closing Databricks connection : " + e.getMessage());
+                log.debug("Error closing Databricks connection : " + e.getMessage());
```
```suggestion
                log.error("Error closing Databricks connection : " + e.getMessage());
```

AnaghHegde commented on app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java:
```diff
@@ -952,12 +952,12 @@ public Mono<Firestore> datasourceCreate(DatasourceConfiguration datasourceConfig
                     connection.listCollections();
                 } catch (FirestoreException e) {
-                    System.out.println("Invalid datasource configuration: " + e.getMessage());
+                    log.debug("Invalid datasource configuration: " + e.getMessage());
```
```suggestion
                    log.error("Invalid datasource configuration: " + e.getMessage());
```

Please user log.error when dealing with exception.

NilanshBansal commented:
@AnnaHariprasad5123 were you able to work on the changes suggested? And also fix the logging issue?

AnnaHariprasad5123 commented:
I changed all the logs. Now, I am checking the issue

NilanshBansal commented:
> I changed all the logs. Now, I am checking the issue

Thank You, can you please commit the changes done until now, this will help us to review the rest of the things while you work on resolving the logger issue 🙏

AnnaHariprasad5123 commented:
Hi @NilanshBansal, log statements are visible in console now. I have added scope as provided for slf4j dependencies in parent pom of appsmith-plugins and removed slf4j exclusion from pf4j dependency.  I have checked some of the plugins with test- configurations action only. Could you check all logs in all plugins from your end.

**MySQL Plugin**
![image](https://github.com/user-attachments/assets/ce572e9c-4de2-4078-845d-c1a934adb80d)
**Postgres Plugin**
![image](https://github.com/user-attachments/assets/d1cf462b-d8d8-494f-8e0a-ec7eb714c7a9)
**Amazon S3 Plugin** 
![image](https://github.com/user-attachments/assets/1a06341c-885b-4213-917a-269fee0c4fd1)
**Dynamo Plugin**
![image](https://github.com/user-attachments/assets/fd7b795f-91a9-46ae-b2e9-2f74cc1e9751)

NilanshBansal commented:
Thank You @AnnaHariprasad5123. I am checking this

NilanshBansal commented:
Thanks @AnnaHariprasad5123 for your valuable contribution! We highly appreciate your efforts in making Appsmith better ❤️
You can find more issues to contribute from [this](https://github.com/appsmithorg/appsmith/issues?q=is%3Aopen+is%3Aissue+label%3A%22Inviting+Contribution%22) list!
